### ISSUE TITLE
Wrong error when trying to write to static in a static initializer

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#51251]
http://play.rust-lang.org/?gist=d719594f15f35f8be13987e2f722d1ab&version=nightly&mode=debug

```rust
#![feature(const_let)]

static mut FOO: usize = unsafe {
    FOO = 42;
    5
};

fn main() {
}
```

results in

```
error[E0658]: statements in statics are unstable (see issue #48821)
 --> src/main.rs:4:5
  |
4 |     FOO = 42;
  |     ^^^^^^^^
  |
  = help: add #![feature(const_let)] to the crate attributes to enable
```

even though the feature gate is active (it should still error, but with a more targeted error message)
